pkg/importer/dataStream.go

```diff
@@ -467,8 +483,18 @@ func closeReaders(readers []Reader) (rtnerr error) {
 	return rtnerr
 }
 
+func (d dataStream) isHTTPQcow2() bool {
+	// assuming len(d.Readers) == 3 is [http, mr, mr]
+	// should prob get rid of the redundant MultiReader
+	return (d.Url.Scheme == "http" || d.Url.Scheme == "https") && d.qemu && len(d.Readers) == 3
```

Comment:
`== 3`? vs. `>=3` (or 2 if mr fix). It looks like we only invoke streaming if the endpoint has not been compressed and/or archived, correct? I thought that http(s) supported some form of inline decompression and, if true, is this something we can use?

Reply:
Yeah, qemu-img can't work directly on compressed/tarred files.

I'm not aware of any http features that can help us here.    An http client can request that the server compress data for transmission via the `Accept-Encoding` header.  But I'm not aware of any protocol support for decompressing a request for a gzipped file.

Reply:
Also, `qemu-img convert` doesn't support reading from stdin so we can't pipe data into it from a qzip reader.  If the file is compressed/tarred, I think we have to copy it locally,